test/e2e/user_management_test.go

```diff
@@ -42,7 +45,99 @@ func (s *userManagementTestSuite) TearDownTest() {
 	s.testCtx.Cleanup()
 }
 
+func (s *userManagementTestSuite) TestUserDeactivation() {
+	s.setApprovalPolicyConfig("automatic")
+
+	userSignup, mur := s.createAndCheckUserSignup(true, "iris-at-redhat-com", "iris@redhat.com", approvedByAdmin()...)
+
+	// Deactivate the user
+	userSignup.Spec.Deactivated = true
+	err := s.awaitility.Client.Update(context.TODO(), userSignup)
+	require.NoError(s.T(), err)
+	s.T().Logf("user signup '%s' set to deactivated", userSignup.Name)
+
+	err = s.hostAwait.WaitUntilMasterUserRecordDeleted(mur.Name)
+	require.NoError(s.T(), err)
+}
+
+func (s *userManagementTestSuite) TestUserBanning() {
+
+	s.T().Run("user banning", func(t *testing.T) {
+		// when
+		s.setApprovalPolicyConfig("automatic")
+
+		// then
+		s.checkUserBanned()
+	})
+
+}
+
+func (s *userManagementTestSuite) checkUserBanned() {
+	s.T().Run("ban provisioned usersignup", func(t *testing.T) {
+		s.setApprovalPolicyConfig("automatic")
+
+		// Create a new UserSignup and confirm it was approved automatically
+		userSignup, _ := s.createUserSignupAndAssertAutoApproval(false)
+
+		// Create the BannedUser
+		s.createAndCheckBannedUser(userSignup.Annotations[v1alpha1.UserSignupUserEmailAnnotationKey])
+
+		// Confirm the user is banned
+		_, err := s.hostAwait.WithRetryOptions(wait.TimeoutOption(time.Second*10)).WaitForUserSignup(userSignup.Name,
+			wait.UntilUserSignupHasConditions(approvedAutomaticallyAndBanned()...))
+		require.NoError(s.T(), err)
+
+		// Confirm that a MasterUserRecord is deleted
+		_, err = s.hostAwait.WithRetryOptions(wait.TimeoutOption(time.Second * 10)).WaitForMasterUserRecord(userSignup.Spec.Username)
+		require.Error(s.T(), err)
+	})
+
+	s.T().Run("create usersignup with preexisting banneduser", func(t *testing.T) {
+		s.setApprovalPolicyConfig("automatic")
+
+		id := uuid.NewV4().String()
+		email := "testuser" + id + "@test.com"
+		s.createAndCheckBannedUser(email)
+
+		_ = s.createAndCheckUserSignupNoMUR(false, "testuser"+id, email, banned()...)
```

Comment:
here, you only check that the UserSignup exists with the given condition, but not that the MUR doesn't exist - you should add that check too

Reply:
That's not so simple... the MUR isn't ever created in the first place, and so a compliant name is never generated for it and therefore I don't even have an MUR name to use for the check.  Do you have any ideas for how this can be easily checked?

Reply:
yes, by listing all MURs. 
You can also use the label selector for the `toolchain.dev.openshift.com/user-id` label equal to `UserSignup.name`

Reply:
Actually, in this particular case the username is enough to check the MUR. It's a regular username (testuser + UUID), so it would match the compliant name.
I've added the check in https://github.com/codeready-toolchain/toolchain-e2e/pull/80/commits/47aff7e47cb17aa3ae92efe2064202c4b3837775